oasislmf/utils/data.py

```diff
@@ -547,6 +548,12 @@ def get_dataframe(
         if missing:
             raise OasisException('Missing required columns: {}'.format(missing))
 
+    # Force categorical column values to string
```

Comment:
I'm not sure if this is useful, I don't see change in the dtype of the category
>>> import pandas as pd
>>> import io
>>> a = io.StringIO("a,b,c\n1,a,1\n2,a,1")
>>> df = pd.read_csv(a, dtype={"b": 'category', 'c':'category'})
>>> df["c"].cat.categories
Index(['1'], dtype='object')
>>> df["c"].cat.categories = df["c"].cat.categories.astype(str)
>>> df["c"].cat.categories
Index(['1'], dtype='object')
I didn't see it forbid int values for example
>>> df['c'].cat.add_categories([2], inplace=True)
>>> df['c'][0]=2
>>> df
   a  b  c
0  1  a  2
1  2  a  1

Reply:
I'll try to remind myself of where this turned out to be a problem. The reasoning is that we want the category values to be string - there's been plenty of fun where one column is of type "int" and the other "string", and Pandas won't merge `1` with `"1"`. However, I recall this being a rather awkward case that didn't always occur - as in your example, `type(df['c'][0].iat[0])` returns `<class 'str'>`.

(It may have been a complication of Oasis replacing special/null values with OED defaults like `1`)

Reply:
Amusingly, I've not been able to reproduce any problem by removing these lines. I suspect that this was working around another issue.